Libraries/Azure.psm1

```diff
@@ -25,164 +25,181 @@
-		$AllowedUsagePercentage = 100
-
-		#Get the region
-		$currentStatus = Get-AzVMUsage -Location $Location
-		$overFlowErrors = 0
-		$vmCounter = 0
-		foreach ($VM in $RGXMLData.VirtualMachine) {
-			$vmCounter += 1
-			Write-LogInfo "Estimating VM #$vmCounter usage."
-			# this 'OverrideVMSize' is already one item from OverrideVMSize Array that already splitted by ','
-			if ($OverrideVMSize) {
-				$testVMSize = $OverrideVMSize
+		if (!$AllTestVMSizes.$testVMSize) {
+			Measure-SubscriptionCapabilities
+		}
+		if ($AllTestVMSizes.$testVMSize) {
+			$vmCPUs = $global:AllTestVMSizes.$testVMSize.vCPUs
+			$vmFamily = $global:AllTestVMSizes.$testVMSize.Family
+			if ($vmFamilyRequiredCPUs.$vmFamily) {
+				$vmFamilyRequiredCPUs.$vmFamily += $vmCPUs
 			}
 			else {
-				$testVMSize = $VM.ARMInstanceSize
+				$vmFamilyRequiredCPUs["$vmFamily"] = $vmCPUs
 			}
 
-			if ($OverrideVMSize -and ($testVMUsage -gt 0)) {
-				#Do nothing.
+			if (!$TargetLocation.Value) {
+				$vmAvailableLocations = $global:AllTestVMSizes.$testVMSize.AvailableLocations
 			}
 			else {
-				$testVMUsage = (Get-AzVMSize -Location $Location | Where-Object { $_.Name -eq $testVMSize }).NumberOfCores
-			}
-
-			$DetectedVMFamily = ($ComputeSKU | Where-Object { $_.Name -eq $testVMSize }).Family | Get-Unique
-			if ($DetectedVMFamily) {
-				$currentStatus = Set-Usage -currentStatus $currentStatus -VMFamily $DetectedVMFamily  -usage $testVMUsage -AllowedUsagePercentage $AllowedUsagePercentage
-				$overFlowErrors += Test-Usage -currentStatus $currentStatus -VMFamily $DetectedVMFamily -AllowedUsagePercentage $AllowedUsagePercentage
+				$vmAvailableLocations = @($TargetLocation.Value.Trim(", ").Split(',').Trim())
+			}
+			$index = 0
+			for ($index = 0; $index -lt $vmAvailableLocations.Count; ) {
+				# 'cores' is the overall region allowed vCPUs
+				$regionVmUsage = Get-AzVMUsage -Location $vmAvailableLocations[$index]
+				$vmFamilyUsage = $regionVmUsage | Where-Object { $_.Name.Value -imatch "$vmFamily" } | Select-Object CurrentValue, Limit
+				$regionCoresUsage = $regionVmUsage | Where-Object { $_.Name.Value -imatch "$vmFamily" } | Select-Object CurrentValue, Limit
+				$locationErrors = Test-OverflowErrors -ResourceType "$vmFamily" -CurrentValue $vmFamilyUsage.CurrentValue `
+					-RequiredValue $vmFamilyRequiredCPUs.$vmFamily -MaximumLimit $vmFamilyUsage.Limit -Location $vmAvailableLocations[$index]
+				$locationErrors += Test-OverflowErrors -ResourceType "Region Total Cores" -CurrentValue $regionCoresUsage.CurrentValue `
+					-RequiredValue $vmFamilyRequiredCPUs.$vmFamily -MaximumLimit $regionCoresUsage.Limit -Location $vmAvailableLocations[$index]
+				$locationErrors += (&$TestNetworkUsage -NetworkLocation $vmAvailableLocations[$index])
+				if ($locationErrors -gt 0) {
+					$index++
+				}
+				else {
+					Write-LogInfo "Selected Test Location '$vmAvailableLocations[$index]' for Azure Platform testing on VM Size '$testVMSize'"
```

Comment:
Selected Test Location 'westus2[0]' for Azure Platform testing on VM Size 'Standard_DS1_v2', it is expected output?